rest-api/dao/base_dao.py

```diff
@@ -328,19 +327,23 @@ def _insert_with_random_id(self, obj, fields):
       try:
         with self.session() as session:
           return self.insert_with_session(session, obj)
-      except IntegrityError, e:
-        # SQLite and MySQL variants of the error message, respectively.
-        if 'UNIQUE constraint failed' in e.message or 'Duplicate entry' in e.message:
-          if 'external_id' in e.message:
-            raise Conflict('External ID already in use, error: {}'.format(e.message))
-          logging.warning('Failed insert with %s: %s', tried_ids, e.message)
+      except IntegrityError as e:
+        errors = self.handle_integrity_error(tried_ids, e, obj)
```

Comment:
I wouldn't call this errors... maybe result?

Reply:
done